search/tasks.py

```diff
@@ -0,0 +1,108 @@
+from celery import shared_task
+from django.db.utils import DatabaseError
+from elasticsearch.exceptions import TransportError
+from elasticsearch.helpers import bulk
+from elasticsearch_dsl.connections import connections
+from dips.models import Collection, DIP, DigitalFile
+
+import logging
+
+# Use a normal logger to avoid redirecting both `stdout` and `stderr` to the
+# logger and back when using Celery's `get_task_logger`, and to avoid changing
+# the default `CELERY_REDIRECT_STDOUTS_LEVEL` when using `print`.
+logger = logging.getLogger("search.tasks")
+
+
+@shared_task(
+    autoretry_for=(TransportError, DatabaseError),
+    max_retries=10,
+    default_retry_delay=30,
+    ignore_result=True,
+)
+def update_es_descendants(class_name, pk):
```

Comment:
I had some conflicts about this too when I was doing it. Currently, the search tasks are called in the save and delete methods from the parent AbstractEsModel, which should make adding a new model with the same requirements easier. I should have moved the script declaration and other model/doc dependencies from those tasks to the actual model/doc to make it better, but I didn't expect that many differences and it ended all there ;(

About a single task to update each file (I'm not sure if that's what you're suggesting), I thought that, even in an async. context, it would be better to hit Elasticsearch as little as possible, using bulk requests.